### PR TITLE
refactored renderers to their own files and updated import statements

### PR DESCRIPTION
**Pull request by chrilehner**
related issues:   

### Pre merge/review checklist
* [ ] branch is up-to-date with the branch to be merged with, i.e. master
* [ ] travis build works
* [ ] internal code review by requester is done:
  * [ ] code is formatted
  * [ ] removed debug output `console.log`, ...
  * [ ] use of `readonly` for readonly fields and interfaces fields
  * [ ] parameters, fields, variables that are not initialized are typed, e.g.,  `function f(a)` => `function f(a: number)` 
  * [ ] optional options objects are typed using option interfaces, see e.g. `ILineUpConfig`
  * [ ] code is formatted using your preferred editor formatting tool (e.g. atom beautifier, WebStorm Format Code,...)
  * [ ] use of meaningful names in camelCase especially for public functions and fields
  * [ ] file structure: avoid of large code files containing multiple elements (classes, functions) better to create a nested structure, e.g., `model`
  * [ ] in case of DOM elements: assigning the `id` is avoided or the prefix in the lineup config is prepended
  * [ ] in case of css styles: global classes prepended with `lu-` or within a `lu-` namespace
  * [ ] public fields/function those purpose is not obvious are documented
  * [ ] complex code is documented
  * [ ] code is well structured regarding use of variable names, complexity
  * [ ] public interfaces (e.g. public interface of a class) are minimal, no superfluous fields, functions
  * [ ] complex code is not duplicated. Otherwise justified within the code, why needed
 * [ ] code reviewer is assigned
 * [ ] in case of second review round: review comments are justified or implemented

### CODE REVIEW

File: src/renderers/AnnotationRenderer.ts [hunk 1]
```diff
@@ -0,0 +1,76 @@
+import AnnotateColumn from '../model/AnnotateColumn';
+import {IDOMRenderContext, ICanvasRenderContext} from './RendererContexts';
+import {ISVGCellRenderer, IHTMLCellRenderer} from './IDOMCellRenderers';
+import {IDataRow} from '../provider/ADataProvider';
+import ICanvasCellRenderer from './ICanvasCellRenderer';
+import {showOverlay, clipText} from '../utils';
+
+export default class AnnotationRenderer {
```
Comment (sgratzl) [inline src/renderers/AnnotationRenderer.ts]: why is this class not implementing `implements ICellRendererFactory`

File: src/renderers/CircleCellRenderer.ts [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+import Column from '../model/Column';
+import {INumberColumn} from '../model/NumberColumn';
+import {IDOMRenderContext, ICanvasRenderContext} from './RendererContexts';
+import {ISVGCellRenderer} from './IDOMCellRenderers';
+import {IDataRow} from '../provider/ADataProvider';
+import {attr, clipText} from '../utils';
+import ICanvasCellRenderer from './ICanvasCellRenderer';
+
+export default class CircleCellRenderer implements ICellRendererFactory {
+  private readonly renderValue;
```
Comment (sgratzl) [inline src/renderers/CircleCellRenderer.ts]: missing type or add to constructur

File: src/renderers/Heatmap.ts [hunk 3]
```diff
@@ -0,0 +1,69 @@
+import {INumberColumn} from '../model/NumberColumn';
+import Column from '../model/Column';
+import {IDOMRenderContext, ICanvasRenderContext} from './RendererContexts';
+import {ISVGCellRenderer, IHTMLCellRenderer} from './IDOMCellRenderers';
+import {IDataRow} from '../provider/ADataProvider';
+import {attr} from '../utils';
+import ICanvasCellRenderer from './ICanvasCellRenderer';
+import {hsl} from 'd3';
+
+export default class Heatmap {
```
Comment (sgratzl) [inline src/renderers/Heatmap.ts]: why is this class not implementing `implements ICellRendererFactory`

File: src/renderers/Heatmap.ts [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+  private toHeatMapColor(d: any, index: number, col: INumberColumn & Column) {
```
Comment (sgratzl) [inline src/renderers/Heatmap.ts]: can be declared static

File: src/renderers/HeatmapCellRenderer.ts [hunk 5]
```diff
@@ -0,0 +1,73 @@
+import ICellRendererFactory from './ICellRendererFactory';
+import MultiValueColumn from '../model/MultiValueColumn';
+import {IDOMRenderContext, ICanvasRenderContext} from './RendererContexts';
+import {ISVGCellRenderer, IHTMLCellRenderer} from './IDOMCellRenderers';
+import {IDataRow} from '../provider/ADataProvider';
+import ICanvasCellRenderer from './ICanvasCellRenderer';
+import {select as d3select} from 'd3';
+
+
+export default class HeatmapCellRenderer implements ICellRendererFactory {
```
Comment (sgratzl) [inline src/renderers/HeatmapCellRenderer.ts]: since this type will be registered as `multiValue` the class and file name should be named the same

File: src/renderers/LoadingCellRenderer.ts [hunk 6]
```diff
@@ -0,0 +1,39 @@
+import Column from '../model/Column';
+import {ISVGCellRenderer, IHTMLCellRenderer} from './IDOMCellRenderers';
+import {ICanvasRenderContext} from './RendererContexts';
+import ICanvasCellRenderer from './ICanvasCellRenderer';
+import {IDataRow} from '../provider/ADataProvider';
+import {clipText} from '../utils';
+
+export default class LoadingCellRenderer {
```
Comment (sgratzl) [inline src/renderers/LoadingCellRenderer.ts]: why is this class not implementing `implements ICellRendererFactory`

File: src/renderers/SelectionRenderer.ts [hunk 7]
```diff
@@ -0,0 +1,43 @@
+import SelectionColumn from '../model/SelectionColumn';
+import {ISVGCellRenderer, IHTMLCellRenderer} from './IDOMCellRenderers';
+import {IDataRow} from '../provider/ADataProvider';
+import {ICanvasRenderContext} from './RendererContexts';
+import ICanvasCellRenderer from './ICanvasCellRenderer';
+import {clipText} from '../utils';
+
+export default class SelectionRenderer {
```
Comment (sgratzl) [inline src/renderers/SelectionRenderer.ts]: why is this class not implementing `implements ICellRendererFactory`

File: src/ui/ABodyRenderer.ts [hunk 8]
```diff
@@ -99,7 +101,7 @@ abstract class ABodyRenderer extends AEventDispatcher implements IBodyRenderer {
     animation: false, //200
     animationDuration: 1000,
 
-    renderers: merge({}, defaultRenderers),
+    //renderers: merge({}, defaultRenderers),
```
Comment (sgratzl) [inline src/ui/ABodyRenderer.ts]: why is this commented out?

File: src/ui/ADOMBodyRenderer.ts [hunk 9]
```diff
@@ -158,7 +159,7 @@ abstract class ABodyDOMRenderer extends ABodyRenderer {
 
   select(dataIndex: number, additional = false) {
     const selected = super.select(dataIndex, additional);
-    this.$node.selectAll(`[data-data-index="${dataIndex}"`).classed('selected', selected);
+    this.$node.selectAll(`[data-data-index='${dataIndex}'`).classed('selected', selected);
```
Comment (sgratzl) [inline src/ui/ADOMBodyRenderer.ts]: what is the motiviation for this change?

Comment (sgratzl) [inline src/ui/ADOMBodyRenderer.ts]: + there is a bug, the ']' is missing

Comment (chrilehner) [inline src/ui/ADOMBodyRenderer.ts]: sorry, this was an accident

File: src/ui/ABodyRenderer.ts (hunk 8 above)
Comment (chrilehner) [inline src/ui/ABodyRenderer.ts]: I was thinking of another change and forgot to undo this change.
